### ISSUE TITLE
ml.py Bug Report

### ISSUE THREAD

**Issue by Or4cl3AI** [Or4cl3AI/A.I.D.E.N.#20]
Here are some potential bugs and issues in the provided code:

1. The `train` method modifies the `data` and `target` inputs directly by assigning them to `X` and `y`. This could cause issues if the original data and target variables are needed later. To avoid this, create new variables for the training and testing splits.
2. The `train` method calculates the accuracy of the model on the test set, but it does not return the trained model or the test set predictions. This makes it difficult to evaluate the model further or use it for predictions.
3. The `predict` method does not check if the model has been trained before making predictions. If the model has not been trained, this will result in a `ValueError`.
4. The `save_model` and `load_model` methods use `joblib` but do not import it at the beginning of the script. This will result in a `NameError`.
5. The `load_model` method overwrites the current model with the loaded model, but it does not update the `self.model` attribute of the class. This means that the loaded model will not be used for future predictions.
6. The code does not handle any exceptions or errors that might occur during training, prediction, or model saving/loading. Adding some error handling could make the code more robust and easier to debug.

Here's an updated version of the code that addresses these issues:

```python
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
import joblib
import pandas as pd

class ML:
    def __init__(self):
        self.model = RandomForestClassifier()

    def train(self, data, target):
        X_train, X_test, y_train, y_test = train_test_split(data, target, test_size=0.2, random_state=42)

        self.model.fit(X_train, y_train)

        y_pred = self.model.predict(X_test)

        accuracy = accuracy_score(y_test, y_pred)

        return self.model, accuracy, y_pred
